src/components/KCodeBlock/KCodeBlock.vue

```diff
@@ -780,491 +710,217 @@ async function copyCode(event: Event): Promise<void> {
+      &.single-line {
+        grid-template-columns: auto;
+        padding-right: var(--kui-space-100, $kui-space-100);
+      }
+    }
 
-  .theme-light {
-    color: var(--kui-color-text-neutral, $kui-color-text-neutral);
+    .code-block-secondary-actions {
+      display: flex;
+      gap: var(--kui-space-40, $kui-space-40);
+      margin-right: var(--kui-space-40, $kui-space-40);
+      margin-top: var(--kui-space-40, $kui-space-40);
+      position: absolute;
+      right: 0;
+      top: 0;
+      z-index: 1;
+    }
   }
 
-  .theme-dark {
-    color: var(--kui-color-text-neutral-weak, $kui-color-text-neutral-weak);
-  }
-}
+  &.theme-dark {
+    background-color: var(--kui-color-background-inverse, $kui-color-background-inverse);
+
+    .code-block-actions {
+      border-bottom-color: var(--kui-color-border-inverse, $kui-color-border-inverse);
+
+      .code-block-search-input {
+        :deep(input) {
+          color: var(--kui-color-text-inverse, $kui-color-text-inverse);
+
+          &::placeholder {
+            color: var(--kui-color-text-neutral-weaker, $kui-color-text-neutral-weaker);
+          }
+        }
+
+        .clear-query-button,
+        .code-block-search-icon {
+          color: var(--kui-color-text-neutral-weaker, $kui-color-text-neutral-weaker) !important;
+        }
+
+        .clear-query-button[role="button"] {
+          &:hover, &:focus, &:focus-visible {
+            color: var(--kui-color-text-inverse, $kui-color-text-inverse) !important;
+          }
+        }
+      }
 
-.k-clear-query-button {
-  align-items: center;
-  appearance: none;
-  background-color: var(--kui-color-background-transparent, $kui-color-background-transparent);
-  border: var(--kui-border-width-10, $kui-border-width-10) solid var(--kui-color-border-transparent, $kui-color-border-transparent);
-  border-radius: var(--kui-border-radius-10, $kui-border-radius-10);
-  color: var(--kui-color-text-neutral-weak, $kui-color-text-neutral-weak);
-  display: inline-flex;
-  font: inherit;
-  margin: var(--kui-space-0, $kui-space-0);
-  padding: var(--kui-space-0, $kui-space-0) var(--kui-space-20, $kui-space-20);
-
-  .k-clear-icon {
-    .theme-light {
-      color: var(--kui-color-text-neutral, $kui-color-text-neutral);
+      .code-block-search-results {
+        color: var(--kui-color-text-inverse, $kui-color-text-inverse);
+      }
     }
 
-    .theme-dark {
-      color: var(--kui-color-text-neutral-weak, $kui-color-text-neutral-weak);
+    .code-block-content {
+      pre {
+        .line-number-rows {
+          .line {
+            .line-anchor {
+              color: var(--kui-color-text-neutral-weak, $kui-color-text-neutral-weak);
+            }
+
+            &.line-is-match {
+              &::before {
+                background-color: $tmp-color-white-opaque-12; // we don't have a token for this now and we might not ever need one so using a temp variable
```

Comment:
@adamdehaven fyi I decided to use tmp variable for this value

Reply:
sgtm